src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -554,6 +551,20 @@ internal ExitWriteLock DrainRunningQueries()
             }
         }
 
+        private void InitializeComponentsUsingEnvironment(DocumentDatabase documentDatabase, StorageEnvironment environment)
+        {
+            _contextPool?.Dispose();
+            _contextPool = new TransactionContextPool(environment);
+            
+            _indexStorage = new IndexStorage(this, _contextPool, documentDatabase);
```

Comment:
You aren't calling `_indexStorage ?.Dispose()` intentionally?

Reply:
It is not disposable, it just wraps _environment